Intermittent failure of database functional test

As seen at http://ci.deis.io/job/test-integration/142/console, test failure on `TestDatabase`:

```
=== RUN TestDatabase
--- Running deis/test-etcd at 10.0.1.15:51172
Error response from daemon: No such container: deis-etcd-jenkins-test-integration-142
[etcd] Sep 29 18:50:15.090 WARNING   | Using the directory 7593729e3d5f.etcd as the etcd curation directory because a directory was not specified. 
[etcd] Sep 29 18:50:15.091 INFO      | 7593729e3d5f is starting a new cluster
[etcd] Sep 29 18:50:15.092 INFO      | etcd server [name 7593729e3d5f, listen on :51172, advertised url http://10.0.1.15:51172]
[etcd] Sep 29 18:50:15.092 INFO      | peer server [name 7593729e3d5f, listen on :7001, advertised url http://127.0.0.1:7001]
[etcd] Sep 29 18:50:15.092 INFO      | 7593729e3d5f starting in peer mode
[etcd] Sep 29 18:50:15.092 INFO      | 7593729e3d5f: state changed from 'initialized' to 'follower'.
[etcd] Sep 29 18:50:15.092 INFO      | 7593729e3d5f: state changed from 'follower' to 'leader'.
[etcd] Sep 29 18:50:15.092 INFO      | 7593729e3d5f: leader changed from '' to '7593729e3d5f'.
-v test
Getting inspect data :{{ .Config.Hostname }}:deis-database-data
4918170e3f73
data container 4918170e3f73
--- Run deis/database:jenkins-test-integration-142 at 10.0.1.15:45368
Error response from daemon: No such container: deis-database-jenkins-test-integration-142
2014-09-29T18:50:31Z 3857f1f10781 confd[43]: ERROR exit status 127
2014-09-29T18:50:31Z 3857f1f10781 confd[43]: ERROR exit status 127
database: waiting for confd to write initial templates...
2014-09-29 18:50:36 UTC FATAL:  data directory "/var/lib/postgresql/9.3/main" has wrong ownership
2014-09-29 18:50:36 UTC HINT:  The server must be started by the user that owns the data directory.
2014-09-29T19:00:17Z 3857f1f10781 confd[81]: ERROR 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
2014-09-29T19:00:17Z 3857f1f10781 confd[81]: ERROR 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
2014-09-29T19:00:17Z 3857f1f10781 confd[81]: ERROR 501: All the given peers are not reachable (Tried to connect to each peer twice and failed) [0]
*** Test killed: ran too long (10m0s).
FAIL    github.com/deis/deis/database/tests 605.024s
make[1]: *** [test-functional] Error 1
make[1]: Leaving directory `/home/jenkins/workspace/test-integration/src/github.com/deis/deis/database'
make: *** [test-components] Error 2

>>> Error detected, dumping logs <<<
```

--- a/database/tests/database_test.go
+++ b/database/tests/database_test.go
@@ -17,7 +17,8 @@ func TestDatabase(t *testing.T) {
 	cli, stdout, stdoutPipe := dockercli.NewClient()
 	dockercli.RunTestEtcd(t, etcdName, etcdPort)
 	defer cli.CmdRm("-f", etcdName)
-	dockercli.RunDeisDataTest(t, "--name", "deis-database-data",
+	dataName := "deis-database-data-" + tag
+	dockercli.RunDeisDataTest(t, "--name", dataName,
 		"-v", "/var/lib/postgresql", "deis/base", "true")
 	host, port := utils.HostAddress(), utils.RandomPort()
 	fmt.Printf("--- Run deis/database:%s at %s:%s\n", tag, host, port)
@@ -32,7 +33,7 @@ func TestDatabase(t *testing.T) {
 			"-e", "EXTERNAL_PORT="+port,
 			"-e", "HOST="+host,
 			"-e", "ETCD_PORT="+etcdPort,
-			"--volumes-from", "deis-database-data",
+			"--volumes-from", dataName,
 			"deis/database:"+tag)
 	}()
 	dockercli.PrintToStdout(t, stdout, stdoutPipe, "deis-database running")

--- a/logger/tests/logger_test.go
+++ b/logger/tests/logger_test.go
@@ -17,7 +17,8 @@ func TestLogger(t *testing.T) {
 	cli, stdout, stdoutPipe := dockercli.NewClient()
 	dockercli.RunTestEtcd(t, etcdName, etcdPort)
 	defer cli.CmdRm("-f", etcdName)
-	dockercli.RunDeisDataTest(t, "--name", "deis-logger-data",
+	dataName := "deis-logger-data-" + tag
+	dockercli.RunDeisDataTest(t, "--name", dataName,
 		"-v", "/var/log/deis", "deis/base", "/bin/true")
 	host, port := utils.HostAddress(), utils.RandomPort()
 	fmt.Printf("--- Run deis/logger:%s at %s:%s\n", tag, host, port)
@@ -32,7 +33,7 @@ func TestLogger(t *testing.T) {
 			"-e", "EXTERNAL_PORT="+port,
 			"-e", "HOST="+host,
 			"-e", "ETCD_PORT="+etcdPort,
-			"--volumes-from", "deis-logger-data",
+			"--volumes-from", dataName,
 			"deis/logger:"+tag)
 	}()
 	dockercli.PrintToStdout(t, stdout, stdoutPipe, "deis-logger running")

--- a/registry/tests/registry_test.go
+++ b/registry/tests/registry_test.go
@@ -26,7 +26,8 @@ func TestRegistry(t *testing.T) {
 	defer cli.CmdRm("-f", etcdName)
 	handler := etcdutils.InitEtcd(setdir, setkeys, etcdPort)
 	etcdutils.PublishEtcd(t, handler)
-	dockercli.RunDeisDataTest(t, "--name", "deis-registry-data",
+	dataName := "deis-registry-data-" + tag
+	dockercli.RunDeisDataTest(t, "--name", dataName,
 		"-v", "/data", "deis/base", "/bin/true")
 	host, port := utils.HostAddress(), utils.RandomPort()
 	fmt.Printf("--- Run deis/registry:%s at %s:%s\n", tag, host, port)
@@ -41,7 +42,7 @@ func TestRegistry(t *testing.T) {
 			"-e", "EXTERNAL_PORT="+port,
 			"-e", "HOST="+host,
 			"-e", "ETCD_PORT="+etcdPort,
-			"--volumes-from", "deis-registry-data",
+			"--volumes-from", dataName,
 			"deis/registry:"+tag)
 	}()
 	dockercli.PrintToStdout(t, stdout, stdoutPipe, "Booting")

